### ISSUE TITLE
Additional md5 check to reduce corrupt images

### ISSUE THREAD

**Issue by ToksT** [r888888888/danbooru#2569]
Cases where an image is corrupt but the md5 is that of the uncorrupt image ([example](http://danbooru.donmai.us/posts/2239030)) should be possible to detect and raise an error. Right at the end of upload processing (before calling post.save) recalculate the md5, and if it doesn't match the md5 that was calculated earlier then the post must have become corrupt during upload.

By the way, the most recent corrupt images have duplicate entries in the upload log. So I'd guess the cause is probably when multiple people try to upload the same image at the same time, multiple upload processes will try to move the same image at the same time.

**Comment by r888888888** [comment]
I realized that the ftp distribute could cause issues. It's possible that's failing and leaving a bad file on other servers.

**Comment by ToksT** [comment]
For some reason several uploads have had this error since the change was made

```
An error occurred: error: Net::SFTP::StatusException - Net::SFTP::StatusException (4, "failure")

    /var/www/danbooru2/shared/bundle/ruby/2.1.0/gems/net-sftp-2.1.2/lib/net/sftp/session.rb:846:in `wait_for'
    /var/www/danbooru2/shared/bundle/ruby/2.1.0/gems/net-sftp-2.1.2/lib/net/sftp/session.rb:590:in `rename!'
    /var/www/danbooru2/releases/20160119021851/app/logical/remote_file_manager.rb:17:in `block (2 levels) in distribute'
```

It seems that once a given image has had that error, any future attempts to upload the same image will get the same error, so you can't just retry.

**Comment by r888888888** [comment]
I believe this is just an error that pops up when the file already exists, likely when someone else tried to upload it at around the same time. I'll reprocess the failed uploads that have sources but the ones that were uploaded by file will have to be deleted and retried.

**Comment by r888888888** [comment]
Fix in 802b31a34e2cea015707ec14d7428d0517f28111
